miso-web/src/main/webapp/scripts/library_hot.js

```diff
@@ -476,108 +493,112 @@ Library.hot = {
+      lib.libraryTypeId = Hot.getIdFromAlias(obj.libraryTypeAlias, Hot.dropdownRef['libraryTypes']);
+
+      if (obj.lowQuality !== undefined) {
+        lib.lowQuality = obj.lowQuality;
+      } else {
+        lib.lowQuality = false;
+      }
+
       if (obj.tagBarcodeFamilyName && obj.tagBarcodeFamilyName != 'No barcode') {
         var tbfn = obj.tagBarcodeFamilyName;
         lib.tagBarcodeFamilyName = tbfn;
-        if (obj.tagBarcodes.one.alias) {
-          lib.tagBarcodeIndex1Id = Library.hot.getIdFromBcComposite(obj.tagBarcodes.one.alias, Hot.dropdownRef.tagBarcodes[tbfn]['1']);
+        if (obj.tagBarcodeIndex1Label) {
+          lib.tagBarcodeIndex1Id = Library.hot.getIdFromBcLabel(obj.tagBarcodeIndex1Label, Hot.dropdownRef.tagBarcodes[lib.platformName][tbfn]['1']);
+        } else {
+          lib.tagBarcodeIndex1Id = undefined;
         }
-        if (obj.tagBarcodes.two.alias) {
-          lib.tagBarcodeIndex2Id = Library.hot.getIdFromBcComposite(obj.tagBarcodes.two.alias, Hot.dropdownRef.tagBarcodes[tbfn]['2']);
+        if (obj.tagBarcodeIndex2Label) {
+          lib.tagBarcodeIndex2Id = Library.hot.getIdFromBcLabel(obj.tagBarcodeIndex2Label, Hot.dropdownRef.tagBarcodes[lib.platformName][tbfn]['2']);
+        } else {
+          lib.tagBarcodeIndex2Id = undefined;
         }
       }
-    }
-    
-    if (obj.volume) {
-      lib.volume = obj.volume;
-    } else {
-      lib.volume = 0;
-    }
-    
-    if (obj.libraryAdditionalInfo) {
-      lib.libraryAdditionalInfo = {
-        tissueOrigin: obj.libraryAdditionalInfo.tissueOrigin,
-        tissueType: obj.libraryAdditionalInfo.tissueType
-      };
-      if (obj.libraryAdditionalInfo.prepKit.alias) {
-        var prepKitAlias = obj.libraryAdditionalInfo.prepKit.alias;
-        lib.libraryAdditionalInfo.prepKit = Hot.sampleOptions.kitDescriptorsDtos.filter(function (kd) { return (kd.name == prepKitAlias); })[0];
-      }
-      if (obj.libraryAdditionalInfo.archived) {
-        lib.libraryAdditionalInfo.archived = obj.libraryAdditionalInfo.archived;
+
+      if (obj.volume) {
+        lib.volume = obj.volume;
       } else {
-        lib.libraryAdditionalInfo.archived = false;
+        lib.volume = 0;
       }
+
+      if (obj.libraryAdditionalInfo) {
+        lib.libraryAdditionalInfo = {};
+        if (obj.libraryAdditionalInfo.prepKit.alias) {
+          var prepKitAlias = obj.libraryAdditionalInfo.prepKit.alias;
+          lib.libraryAdditionalInfo.prepKit = Hot.sampleOptions.kitDescriptorsDtos.filter(function (kd) { return (kd.name == prepKitAlias); })[0];
+        }
+        if (obj.libraryAdditionalInfo.archived) {
+          lib.libraryAdditionalInfo.archived = obj.libraryAdditionalInfo.archived;
+        } else {
+          lib.libraryAdditionalInfo.archived = false;
+        }
+      }
+
+      lib.qcPassed = obj.qcPassed;
+
+      // TODO: add qcCols
+    } catch (e) {
+      console.log(e.error);
```

Comment:
So long as you will be sure it will definitely break if everything isn't processed. I'm generally a fan of returning something definitely broken in this case in case there's some error in the future. I wouldn't want the user to see a successful save that's eaten half their data.